File: src/main/java/pl/SFG/SGF/model/game/fighting/FightAction.java
package pl.SFG.SGF.model.game.fighting;

import lombok.Data;
import pl.SFG.SGF.dto.fighting.FighterStatsDto;

import java.util.ArrayList;
import java.util.List;

@Data
public class FightAction {

    private FighterStatsDto attacker;

    private FighterStatsDto target;

    private List<Integer> damage = new ArrayList<>();
    private List<Float> attackTime = new ArrayList<>();
    private List<Boolean> crit = new ArrayList<>();

}

File: frontend/SFG_Frontend/src/features/fighting/components/FightAction.ts
import type { FighterStatsDto } from "./FighterStatsDto";

export interface FightAction {
  attacker: FighterStatsDto;
  target: FighterStatsDto;

  targetHpLeft:number[];
  damage: number[];
  attackTime: number[];
  crit: boolean[];
}